.github/workflows/ci.yml

```diff
@@ -0,0 +1,85 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  # Allow manual runs. It is necessary to run the workflow manually, when the "formatter" workflow applies any changes.
+  workflow_dispatch:
+
+jobs:
+  Processor_Partial:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest, macos-13] # macos-13 for x86_x64 arch
+        framework: [net8.0, net9.0]
+        include:
+          - os: windows-latest
+            framework: net462
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup .NET SDK v9.0.x
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 9.0.x
+      - name: Setup .NET SDK v8.0.x (if needed)
+        if: ${{ matrix.framework == 'net8.0' }}
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.x
+      - name: Run tests
+        shell: bash
+        run: dotnet test --configuration=Release -v=Normal --framework=${{ matrix.framework }}
```

Comment:
After having seen the level of detail we are getting when the tests run, I feel like removing `-v=Normal` would actually be beneficial. It's currently just giving us detailed info about the files being built, which is not very relevant.